zerver/lib/outgoing_webhook.py

```diff
@@ -41,23 +41,25 @@ def process_event(self, event):
         # type: (Dict[str, Any]) -> Tuple[Dict[str ,Any], Dict[str, Any]]
         raise NotImplementedError()
 
-    # Given a successful response to the outgoing webhook REST operation, returns the message
-    # that should be sent back to the user.
+    # Given a successful response to the outgoing webhook REST operation, determines if response is required
+    # to be sent to user and in such case, returns the message that should be sent back.
     #
     # The response will be the response object obtained from REST operation.
     # The event will be the same as the input to process_command.
-    # The returned message will be a dictionary which should have "response_message" as key and response message to
-    # be sent to user as value.
+    # The returned message will be a dictionary which should have atleast
+    # 1. "response_required" as key and boolean True or False as value.
```

Comment:
Using string keys in this way works, but it is not ideal for a few reasons:
- It's really easy to make a typo (especially if you don't use string constants for the keys)
- It's really easy to forget a key or add an extraneous key
- The type system (MyPy in this case) can't check for safety

Would it be possible to consider some alternative approaches here?  For example:
- Return `None` if no message should be sent
- Return a tuple with 2 elements, where the 2nd element is a boolean that determines whether or not a response should be sent